mizar/cni/mizarcni.py

```diff
@@ -141,32 +141,38 @@ def activate_interface(self, interface):
         moves the interface to the CNI netnt, rename it, set the IP address, and
         the GW.
         """
-
-        # TODO (cathy): skip the interface activation if an interface with same
-        # interface.interface_id.interface already UP and configured in the
-        # namespace
-
         head, netns = os.path.split(self.netns)
         iproute_ns = NetNS(netns)
         veth_index = get_iface_index(interface.veth.name, self.iproute)
-        logger.error("Move interface {}/{} to netns {}".format(
-            interface.veth.name, veth_index, netns))
-        self.iproute.link('set', index=veth_index, net_ns_fd=netns)
 
-        # configure and activate interfaces inside the netns
-        iproute_ns.link('set', index=veth_index, ifname=self.interface)
-        lo_idx = iproute_ns.link_lookup(ifname="lo")[0]
-        iproute_ns.link('set', index=lo_idx, state='up')
+        # skip the interface activation if an interface with same
+        # interface.interface_id.interface already UP and configured in the
+        # namespace
+        actived_idxs = iproute_ns.link_lookup(operstate="UP")
+        if (veth_index not in actived_idxs):
+            logger.error("Move interface {}/{} to netns {}".format(
+                interface.veth.name, veth_index, netns))
+            self.iproute.link('set', index=veth_index, net_ns_fd=netns)
+
+            # configure and activate interfaces inside the netns
+            iproute_ns.link('set', index=veth_index, ifname=self.interface)
+            lo_idx = iproute_ns.link_lookup(ifname="lo")[0]
+            iproute_ns.link('set', index=lo_idx, state='up')
```

Comment:
Why we do need this here? If the function does not return, the configuration will be done again after the branching. Right?